ci_tests/src/index.ts

```diff
@@ -134,215 +205,327 @@ async function deployEth(
+  //   SOL_PRIVATE_KEY,
+  //   BigInt(10000000)
+  // );
+  // console.log("Minted 10000000 tokens");
+  await spl.setAuthority(
+    SOL_CONNECTION,
+    SOL_PRIVATE_KEY,
+    mint,
+    SOL_PRIVATE_KEY,
+    0, // mint
+    SOL_NTT_CONTRACT.tokenAuthorityAddress()
+  );
+  console.log(
+    "Set token authority to",
+    SOL_NTT_CONTRACT.tokenAuthorityAddress().toString()
+  );
+
+  await SOL_NTT_CONTRACT.initialize({
+    payer: SOL_PRIVATE_KEY,
+    owner: SOL_PRIVATE_KEY,
+    chain: "solana",
+    mint,
+    outboundLimit: new BN(1000000000),
+    mode: "burning",
+  });
+  console.log(
+    "Initialized ntt at",
+    SOL_NTT_CONTRACT.program.programId.toString()
+  );
+
+  await SOL_NTT_CONTRACT.registerTransceiver({
+    payer: SOL_PRIVATE_KEY,
+    owner: SOL_PRIVATE_KEY,
+    transceiver: SOL_NTT_CONTRACT.program.programId,
+  });
+  console.log("Registered transceiver with self");
+
+  return {
+    type: "solana",
+    chainId: 1,
+    chainName: "solana",
+    transceiverAddress: SOL_NTT_CONTRACT.emitterAccountAddress().toString(),
+    managerAddress: SOL_NTT_CONTRACT.program.programId.toString(),
+    NTTTokenAddress: mint.toString(),
+    wormholeCoreAddress: SOL_CORE_ADDRESS,
+    signer: SOL_PRIVATE_KEY,
+  };
+}
+
+async function link(
+  ethInfo: EVMChainDetails,
+  bscInfo: EVMChainDetails,
+  solInfo: SolanaChainDetails
+) {
   // Hook up all the important things together
   /*
     - Manager peer
     - Wormhole peer
     - inbound limits
     https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/evm/test/IntegrationStandalone.t.sol
     */
-  console.log("Starting linking process");
+  console.log("\nStarting linking process");
   console.log("========================");
-  const provider1 = getDefaultProvider(<Networkish>chain1.rpcEndpoint);
-  const signer1 = new Wallet(ETH_PRIVATE_KEY, provider1); // Ganache default private key
-
-  const provider2 = getDefaultProvider(chain2.rpcEndpoint);
-  const signer2 = new Wallet(ETH_PRIVATE_KEY, provider2); // Ganache default private key
 
   const manager1 = NttManager__factory.connect(
-    <string>chain1.managerAddress,
-    signer1
+    ethInfo.managerAddress,
+    ETH_SIGNER
   );
   const manager2 = NttManager__factory.connect(
-    <string>chain2.managerAddress,
-    signer2
+    bscInfo.managerAddress,
+    BSC_SIGNER
   );
 
   const transceiver1 = WormholeTransceiver__factory.connect(
-    <string>chain1.transceiverAddress,
-    signer1
+    ethInfo.transceiverAddress,
+    ETH_SIGNER
   );
   const transceiver2 = WormholeTransceiver__factory.connect(
-    <string>chain2.transceiverAddress,
-    signer2
+    bscInfo.transceiverAddress,
+    BSC_SIGNER
   );
 
   // Would make sense to store the 'client' with a generalized interface instead of the
   console.log("Set manager peers");
-  await (
-    await manager1.setPeer(
-      chain2.chainId,
-      addressToBytes32(chain2.managerAddress),
+  await tryAndWaitThrice(() =>
```

Comment:
Are these done in the proper order for the accountant? I thought that all of the transceivers had to register the hub transceiver first.

Reply:
the order they are done in on-chain doesn't matter, only the order in which they are relayed to the accountant :slightly_smiling_face: